### PR TITLE
Touch interaction: Add a global command to toggle support (Follow-up of #10557)

### PR DESCRIPTION
**Pull request by JulienCochuyt**
### Link to issue number:

Follow-up of PR #10557

### Summary of the issue:

Support of touch interaction can now be disabled thanks to PR #10557.
https://github.com/nvaccess/nvda/pull/10557#issuecomment-622944182 proposed the addition of a dedicated gesture to toggle this setting.

### Description of how this pull request fixes the issue:

Add a global command, bound to `kb:NVDA+shift+t` to toggle touch interaction.

### Known issues with pull request:

As mentioned in https://github.com/nvaccess/nvda/pull/10557#issuecomment-622990734, toggling on the touch interaction support can only be achieved through keyboard or braille gestures.

### Change log entry:

I guess the change log entry for PR #10557 could be augmented to include the new gesture if it is approved before the release.

### CODE REVIEW

**Review by josephsl — DISMISSED** [review]
Er, I'm a new user of NVDA. Is NVDA+Shift+T listed in commands quick reference?

**Comment by josephsl** [comment]
Hi,

Also, when I do git diff between this branch and origin/master, I get submodule commits.

Thanks.

**Comment by JulienCochuyt** [comment]
@josephsl wrote:
> Also, when I do git diff between this branch and origin/master, I get submodule commits.

I'm sorry but I can't find any.
Are you sure your in sync?

**Comment by JulienCochuyt** [comment]
@josephsl wrote:
> Er, I'm a new user of NVDA. Is NVDA+Shift+T listed in commands quick reference?

You're definitely right.
I'm currently traveling. I'll add documentation impact when I get to the hotel tonight.
Thanks.

**Comment by josephsl** [comment]
Hi, yep, they are in sync. Thanks.

**Comment by leonardder** [comment]
this shortcut conflicts with the instant translate layer.

**Comment by lukaszgo1** [comment]
@leonardder wrote:
> 
> this shortcut conflicts with the instant translate layer.


IMO this should not be assigned by default. It is useful only for a tiny minority of users.

**Comment by leonardder** [comment]
Agreed there.

**Review by feerrenrut — CHANGES_REQUESTED** [review]
Can you please add the gesture to the userguide. Under the "Enable touch interaction support" section.

**Comment by JulienCochuyt** [comment]
@leonardder wrote:
> this shortcut conflicts with the instant translate layer.

This add-on indeed binds `nvda+shift+t`
Looking for an alternative, it appears the ["translate" add-on](https://github.com/yplassiard/nvda-translate) by @yplassiard in turns binds `nvda+control+shift+t`.
I'm not aware of the "Instant Translate" user base, but "translate" has already gained some significant popularity in the French community and I'd prefer thus avoid conflicting.

@lukaszgo1 wrote:
> IMO this should not be assigned by default. It is useful only for a tiny minority of users.

I agree there are probably currently more user interested in translation oriented add-ons than they are in this global command.
Still, we can reasonably expect touch interaction to become more of a wider concern in the future, as it seems modern user interfaces keep on evolving this way. (Sure, I miss old text terminals...)
My preferred policy remains on providing default bindings as much as reason allows - It has already proven being a great advantage when providing support and trainings, especially compared to SR like Jaws which comes with localized bindings.

Hence, I'd propose `nvda+alt+t`, unless off course a contrary consensus arises.

**Comment by JulienCochuyt** [comment]
@josephsl wrote:
> Er, I'm a new user of NVDA. Is NVDA+Shift+T listed in commands quick reference?

@feerrenrut wrote:
> Can you please add the gesture to the userguide. Under the "Enable touch interaction support" section.

Both done as of 7eccadb
Note this also fixes the in-doc link to the "Enable touch interaction support" setting as there was an extraneous space between "#" and "TouchSupportEnable" in the anchor specifier.

By the way, we might consider adding more of the default touch gestures to the Commands Quick Reference document, as well as finding some way to move the NVDA Touch Gestures section further down the document.

**Comment by josephsl** [comment]
Hi, does anyone object to assigning Alt+NVDA+T to toggle touch interaction from everywhere? If not, I’ll take a look at this PR this weekend. Thanks.

**Comment by Adriani90** [comment]
nvda+alt+t is already used by the addon time zoner
https://addons.nvda-project.org/addons/timeZoner.en.html

**Comment by Adriani90** [comment]
how about ctrl+alt+nvda+t?

Imo we should discuss if it does not maybe make sense to introduce a second layer for addon? i.e. pressing nvda+alt+spacebar activates the second layer and after that one key stroke will be processed. The second layer could be only dedicated to scripts and commands introduced by addons. The first layer is reserved only for commands in the NVDA core.

**Comment by Adriani90** [comment]
Otherwise at a certain point we will end in a situation where you need more than 5 fingers to execute a command :))

**Comment by lukaszgo1** [comment]
@JulienCochuyt wrote:

> @lukaszgo1 wrote:
> 
> > IMO this should not be assigned by default. It is useful only for a tiny minority of users.
> 
> I agree there are probably currently more user interested in translation oriented add-ons than they are in this global command.
> Still, we can reasonably expect touch interaction to become more of a wider concern in the future, as it seems modern user interfaces keep on evolving this way. (Sure, I miss old text terminals...)

While  the fact that touch screens are getting more and more popular cannot be argued with what makes this command not very interesting for many people is the fact that even on a touch screen pc it adds value only for partially sighted users.

> Hence, I'd propose `nvda+alt+t`, unless off course a contrary consensus arises.

Sounds reasonable.

**Comment by XLTechie** [comment]
>       I agree there are probably currently more user interested in translation oriented add-ons than they are in this global command.
>       Still, we can reasonably expect touch interaction to become more of a wider concern in the future, as it seems modern user interfaces keep on
>       evolving this way. (Sure, I miss old text terminals...)
> 
> While the fact that touch screens are getting more and more popular cannot be argued with what makes this command not very interesting for many people is
> the fact that even on a touch screen pc it adds value only for partially sighted users.
>
>       Hence, I'd propose nvda+alt+t, unless off course a contrary consensus arises.

If its use case is that limited, why not make the default gesture be something 
more obscure, like NVDA+Alt+Ctrl+T, or NVDA+Ctrl+Windows+T.

Using NVDA+Alt+T, keeps that three finger command from being used in the future 
for something more useful to more people, thus forcing something more useful on 
to a more unwieldy gesture.

We should not just default to the next available most simple gesture, but should 
consider how likely most people are to want it there, especially in the light of 
gesture needs of the future.

**Comment by JulienCochuyt** [comment]
@lukaszgo1 wrote:
> what makes this command not very interesting for many people is the fact that even on a touch screen pc it adds value only for partially sighted users.

A blind user might also benefit from this feature in the case of a full screen UI with a usable native touch interface.
We already have different examples of these with musical instruments or game apps in the IOS ecosystem.
I'm sure identical examples for Windows won't last to arise, if not already out there.
I'm limiting this example to full screen, because IIMNW Windows currently lacks the ability to define a control as needing to be kept away from AT touch interception, like IOS does. Btw, we might need in the future to try implement such a concept, to allow app modules to give back proper usability of such UI.

@XLTechie wrote:
> We should not just default to the next available most simple gesture, but should consider how likely most people are to want it there, especially in the light of gesture needs of the future.

Agreed.

> why not make the default gesture be something more obscure, like NVDA+Alt+Ctrl+T, or NVDA+Ctrl+Windows+T. Using NVDA+Alt+T, keeps that three finger command from being used in the future for something more useful to more people, thus forcing something more useful on to a more unwieldy gesture. 

My preference, if we decide for a 4-fingers gesture, would go for `NVDA+control+alt+t`, because:
 - NVDA does not currently have default bindings involving the `windows` key, at least to my knowledge
 - It is easier to type on a keyboard with no left `windows` key, like found on many laptops. Note that the left `alt` key is missing from some keyboard layouts, replaced by `alt gr.` which emulates `control+alt`, leading to a 3-finger gesture.

**Review by feerrenrut — CHANGES_REQUESTED** [review]
Thanks @JulienCochuyt overall this looks good, a minor wording suggestion. Though it would also be good to get feedback from @Qchristensen

**Comment by JulienCochuyt** [comment]
@Qchristensen, waiting for your feedback please regarding wording in the documentation.

@feerrenrut, what do you think regarding the debate on the default gesture? Shall we go for `nvda+control+alt+t`?

**Comment by Qchristensen** [comment]
I agree with Reef's suggestion re changing the wording in globalCommands.py from "Toggles on and off the support of touch interaction" to "Toggles support for touch interaction".  Otherwise, wording all looks good.

File: user_docs/en/userGuide.t2t [hunk 1]
```diff
@@ -1524,6 +1529,7 @@ This category contains the following options:
 This checkbox enables NVDA's touch interaction support.
 If enabled, you can use your fingers to navigate and interact with items on screen using a touchscreen device.
 If disabled, touchscreen support will be disabled as though NVDA is not running.
+This setting can also be toggled on and off using NVDA+alt+t.
```
[the remark this answers is not in the document]
Comment (JulienCochuyt) [reply user_docs/en/userGuide.t2t]: Done as of 371d141.

**Review by feerrenrut — APPROVED** [review]
I think this shortcut is fine.  Thanks for your work on this @JulienCochuyt

**Pull request merged**